github/users.go

```diff
@@ -144,6 +144,22 @@ type UserListOptions struct {
 
 // ListAll lists all GitHub users.
 //
+// To paginate through all users, populate 'Since' with the ID of the last user.
+// Example (warning: this will quickly eat through your GitHub API quota!):
```

Comment:
I did start playing around with inline examples in 32a84d1 and discussed it a little in #323.  One things I haven't liked about them (at least how the current ones are written), is that because they're in the github package, the code is not directly copy and pasteable.  Though now I wonder if we put them in `package github_test`, would that work since we would import the github package like normal.  But would godoc inline the examples with the code if they're in the github_test package?  That might be worth trying out to see, since I actually don't know.